tests/test_internal.py

```diff
@@ -220,11 +220,27 @@ def test_walk_core_schema_copies_ser_schema() -> None:
     resulting in the mutation of the original `__pydantic_core_schema__` dict of a model.
     """
 
+    class MyPlainValidator:
+        """A new change to `PlainValidator` made it so that the issue
+        does not reproduce anymore. This class replicates the old behavior.
+        """
```

Comment:
Hmm, so do we need to test this anymore, if it's fixed?

Reply:
It might be that in another scenario where `return_schema` isn't specified in the wrap serializer schema, the issue still happens. A more robust test would be to assert that generated core schemas for already existing models in a module doesn't change whenever you define a new one. Not sure how easy it is to implement though.

Reply:
Ah yeah, good point. Perhaps then we could remove this test and attempt to add a test for that instead.